src/core/error_helpers.js

```diff
@@ -159,26 +160,53 @@ if (typeof IS_MINIFIED !== 'undefined') {
   // validateParameters() helper functions:
   // lookupParamDoc() for querying data.json
   var lookupParamDoc = function(func) {
```

Comment:
it seems like the rewrites to the lookupParamDoc and testParamType methods could be there own PR, separate from the validation error addition.